### DISCUSSION TITLE
Some PROC's aren't defined.

### DISCUSSION THREAD

**Discussion by tbpaksoy** [ocornut/imgui#8118] [New Users Build/Link/Run issues ONLY!]
Hello when I attempt to complie my code with clang++ I am having this : 

![Code_uyyHefP9is](https://github.com/user-attachments/assets/db41e824-0287-4dc2-96ac-1e48b54bac55)

My code is : 

```

#include <GL/glew.h>

#define IMGUI_DEFINE_MATH_OPERATORS
#include "imgui.h"
#include "imgui.cpp"
#include "imgui_impl_glfw.h"
#include "imgui_impl_glfw.cpp"
#include "imgui_impl_opengl3.h"
#include "imgui_impl_opengl3.cpp"

#include <GLFW/glfw3.h>

int main()
{
    glfwInit();
    glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);

    GLFWwindow *window = glfwCreateWindow(800, 600, "UITest", nullptr, nullptr);
    glfwMakeContextCurrent(window);

    GLenum error = glewInit();

    ImGui::CreateContext();
    ImGui_ImplGlfw_InitForOpenGL(window, true);
    ImGui_ImplOpenGL3_Init("#version 330");
    ImGui::StyleColorsClassic();

    ImGui_ImplGlfw_NewFrame();
    ImGui_ImplOpenGL3_NewFrame();
    ImGui::NewFrame();

    ImGui::Begin("Hello, world!");
    ImGui::Text("This is some useful text.");
    ImGui::End();

    while (!glfwWindowShouldClose(window))
    {

        glfwPollEvents();
        glfwSwapBuffers(window);
    }

    ImGui_ImplOpenGL3_Shutdown();
    ImGui_ImplGlfw_Shutdown();
    ImGui::DestroyContext();

    glfwTerminate();
    return 0;
}

```

and my build task in VS Code is : 

```
"type": "cppbuild",
            "label": "C/C++: clang++.exe build active file",
            "command": "C:\\msys64\\clang64\\bin\\clang++.exe",
            "args": [
                "-fcolor-diagnostics",
                "-fansi-escape-codes",
                "-fsanitize=address",
                "-L${fileDirname}",
                "-lglfw3",
                "-lopengl32",
                "-lglew32",
                "-lfreetype",
                "-lgdi32",
                "-limm32",
                "-g",
                "${file}",
                "-Iinclude",
                "-o",
                "${fileDirname}\\${fileBasenameNoExtension}.exe"
            ],
            "options": {
                "cwd": "${fileDirname}"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": "build",
            "detail": "compiler: C:\\msys64\\clang64\\bin\\clang++.exe"
```

How can I solve this ? Thanks in advance!

**Comment by ocornut** [comment]
> #include <GL/glew.h>
>....
> #include "imgui_impl_opengl3.cpp"

I don't think you can include both glew.h and imgui_impl_opengl3.cpp implementation (which embed its own OpenGL loader) in the same compilation unit, because both will define the same function names.

They will need to be built in separate compilation units.

**Reply by tbpaksoy** [reply]
> > #include <GL/glew.h>
> > ....
> > #include "imgui_impl_opengl3.cpp"
> 
> I don't think you can include both glew.h and imgui_impl_opengl3.cpp implementation (which embed its own OpenGL loader) in the same compilation unit, because both will define the same function names.
> 
> They will need to be built in separate compilation units.

I have removed the `#include <GL/glew.h>` but still getting these : 
![Code_Icxrp33RC9](https://github.com/user-attachments/assets/d855727e-4604-43ab-b6ea-e407b34d9336)

**Reply by ocornut** [reply]
Can you build the example_glfw_opengl3/main.cpp example?

**Reply by tbpaksoy** [reply]
> Can you build the example_glfw_opengl3/main.cpp example?

No, it still gives me same errors.

![Code_U0Ux5poWwY](https://github.com/user-attachments/assets/7f647bb4-db87-412e-ad54-9686032ba4de)

**Reply by tbpaksoy** [reply]
I have tried to build the example file. Maybe it is because of I am using older version of GLEW, is it posible ?

**Reply by tbpaksoy** [reply]
I have update GLEW from 2.1.0 to 2.2.0 but errors still in place.

**Reply by tbpaksoy** [reply]
It seems there are some '#define' and '#ifdef' problems.

![Code_FhaWVBd4z3](https://github.com/user-attachments/assets/8ab292c0-97f9-4703-87c6-b466f2f4fa36)

**Reply by ocornut** [reply]
The example file does not use glew.

**Reply by tbpaksoy** [reply]
When I did `#undef GL_VERSION_1_1` it solved my problem.
